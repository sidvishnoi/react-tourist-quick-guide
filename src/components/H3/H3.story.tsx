import { text, withKnobs, color } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import H3 from '.';

const stories = storiesOf('Atoms/H3', module);
stories.addDecorator(withKnobs);

stories.add('simple', () => {
  return <H3>{text('text', 'This is h3 heading')}</H3>;
});

stories.add('with border', () => {
  return <H3 bordered>{text('text', 'This is h3 heading')}</H3>;
});

stories.add('with color', () => {
  return (
    <H3 bordered color={color('color', 'crimson')}>
      {text('text', 'This is h3 heading')}
    </H3>
  );
});
