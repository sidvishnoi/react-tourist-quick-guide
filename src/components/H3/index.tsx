import * as React from 'react';
import styled from 'styled-components';

const H3 = styled.h3<{ color: string }>`
  --color: ${props => props.color};
  color: var(--color);
  padding: 0.5rem;

  @media print {
    padding: 0 0.5rem 0.2rem;
    --color: #000;
  }
`;

const H3WithBorder = styled(H3)`
  border-bottom: 2px solid var(--color);
`;

export interface H3Props {
  children: React.ReactNode;
  bordered?: boolean;
  color?: string;
}

export default (props: H3Props) => {
  const { bordered = false, color = '#000', children } = props;
  return bordered ? (
    <H3WithBorder color={color}>{children}</H3WithBorder>
  ) : (
    <H3 color={color}>{children}</H3>
  );
};
