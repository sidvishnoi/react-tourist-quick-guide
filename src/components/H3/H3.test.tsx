import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React = require('react');
import * as renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

import H3 from '.';

describe('Component - H3', () => {
  it('renders without border', () => {
    const tree = renderer.create(<H3>PASS</H3>);
    expect(tree).toMatchSnapshot();

    const output = mount(<H3>PASS</H3>);
    expect(output).not.toHaveStyleRule('border-bottom');
  });

  it('renders with border', () => {
    const tree = renderer.create(<H3 bordered>PASS</H3>);
    expect(tree).toMatchSnapshot();

    const output = mount(<H3 bordered>PASS</H3>);
    expect(output).toHaveStyleRule('border-bottom', '2px solid var(--color)');
  });

  it('can change border color', () => {
    const tree = renderer.create(
      <H3 bordered color="red">
        PASS
      </H3>,
    );
    expect(tree).toMatchSnapshot();

    const output = mount(
      <H3 bordered color="red">
        PASS
      </H3>,
    );
    expect(output).toHaveStyleRule('--color', 'red');
  });
});
