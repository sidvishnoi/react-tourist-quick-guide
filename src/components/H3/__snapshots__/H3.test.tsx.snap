// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - H3 can change border color 1`] = `
.c0 {
  --color: red;
  color: var(--color);
  padding: 0.5rem;
  margin: 0;
  border-bottom: 2px solid var(--color);
}

@media print {
  .c0 {
    padding: 0 0.5rem 0.2rem;
    --color: #000;
  }
}

@media print {
  .c0 {
    border-bottom: none;
  }
}

<h3
  className="c0"
  color="red"
>
  PASS
</h3>
`;

exports[`Component - H3 renders with border 1`] = `
.c0 {
  --color: #000;
  color: var(--color);
  padding: 0.5rem;
  margin: 0;
  border-bottom: 2px solid var(--color);
}

@media print {
  .c0 {
    padding: 0 0.5rem 0.2rem;
    --color: #000;
  }
}

@media print {
  .c0 {
    border-bottom: none;
  }
}

<h3
  className="c0"
>
  PASS
</h3>
`;

exports[`Component - H3 renders without border 1`] = `
.c0 {
  --color: #000;
  color: var(--color);
  padding: 0.5rem;
  margin: 0;
}

@media print {
  .c0 {
    padding: 0 0.5rem 0.2rem;
    --color: #000;
  }
}

<h3
  className="c0"
>
  PASS
</h3>
`;
